Realm/RLMAccessor.mm

```diff
@@ -252,6 +286,13 @@ id makeNumberGetter(NSUInteger index, bool boxed, bool optional) {
     }
     return makeGetter<Type>(index);
 }
+template<typename Type>
+id makeWrapperGetter(NSUInteger index, bool optional) {
```

Comment:
Prior to this, there's 5 categories of properties which we supported:
1. Primitive numbers, which don't box at all for non-optional. Optionals are wrapped in util::Optional and converted to NSNumber
2. Types like StringData which are handled identically for optional and non-optional due to being able to store null internally.
3. Lists
4. Object links
5. LinkingObjects

ObjectId doesn't fall into any of these categories: unlike primitive numbers it needs to be boxed for non-optional, and unlike the other non-primitive atomic types it can't represent null and needs to be wrapped in util::Optional for that.